aeron-cluster/src/main/java/io/aeron/cluster/Election.java

```diff
@@ -688,21 +695,45 @@ private int followerLogReplication(final long nowNs)
         else
         {
             workCount += logReplication.doWork();
-            if (logReplication.isDone())
+            if (logReplication.isDone(nowNs))
             {
-                appendPosition = logReplication.position();
-                consensusModuleAgent.logRecordingId(logReplication.recordingId());
-
-                if (consensusPublisher.appendPosition(
-                    leaderMember.publication(), leadershipTermId, appendPosition, thisMember.id()))
+                if (logReplication.isClosed())
                 {
-                    // TODO: only begin replay when commit position from leader is received that matches
-                    cleanupReplication();
-                    state(FOLLOWER_REPLAY, nowNs);
+                    if (leaderLogReplicationCommitPosition >= appendPosition)
+                    {
+                        cleanupReplication();
+                        state(FOLLOWER_REPLAY, nowNs);
+
+                        workCount++;
+                    }
+                    else if (nowNs >= leaderLogReplicationCommitPositionDeadlineNs)
+                    {
+                        throw new ClusterException("Timeout waiting for leader to commit replicated log position");
+                    }
+                    else
+                    {
+                        if (nowNs + ctx.leaderHeartbeatIntervalNs() >= timeOfLastUpdateNs)
+                        {
+                            timeOfLastUpdateNs = nowNs;
+                            consensusPublisher.appendPosition(
```

Comment:
Only update time if the `appendPosition` is successful so it is retried on back pressure.